src/MainOpt.cxx

```diff
@@ -120,119 +112,124 @@ int MainOpt::parse_json_file(string config_file) {
     return -6;
   }
   vali.Reset();
+  set_broker(find_broker(document));
+  broker_config = find_broker_config(document);
+  conversion_threads = find_conversion_threads(document);
+  conversion_worker_queue_size = find_conversion_worker_queue_size(document);
+  main_poll_interval = find_main_poll_interval(document);
```

Comment:
I had to revert this PR because it does not do the right thing here:
If for example `conversion_worker_queue_size` is not specified, then this code will set it to `0` and prevent the forwarder from working at all.
The defaults for these settings are initialized in `MainOpt.h`, and the optiona parsing should only update them if they are specified in some form by the user.

Reply:
Thanks Dominik, makes sense. Will open a new PR with the changes